### PR TITLE
types(GuildChannelManager): correct `fetch` return type

### PR DESCRIPTION
**Pull request by RedGuy12**
**Please describe the changes this PR makes and why it should be merged:**

The [documentation for `/channels/{channelId}`](https://discord.com/developers/docs/resources/channel#get-channel) says:
> Get a channel by ID. Returns a [channel](https://discord.com/developers/docs/resources/channel#channel-object) object. If the channel is a thread, a [thread member](https://discord.com/developers/docs/resources/channel#thread-member-object) object is included in the returned result.

This causes me to believe that this route, and `GuildChannelManager#fetch`, can fetch threads. I tested it and it works.

Note that the bulk fetch type doesn't need to be updated, as [the documentation for that](https://discord.com/developers/docs/resources/guild#get-guild-channels) specifically mentions the exclusion of threads:
> Returns a list of guild [channel](https://discord.com/developers/docs/resources/channel#channel-object) objects. Does not include threads.

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

**Review by Jiralite — CHANGES_REQUESTED** [review]
The documentation needs to be updated too:

https://github.com/discordjs/discord.js/blob/249ba0a9a6a19e9685869f87bf935f3a0f6d55e3/packages/discord.js/src/managers/GuildChannelManager.js#L316-L332

[`GuildChannel`](https://discord.js.org/#/docs/discord.js/main/class/GuildChannel) doesn't include thread channels.

**Comment by RedGuy12** [comment]
> The documentation needs to be updated too:
> 
> [`GuildChannel`](https://discord.js.org/#/docs/discord.js/main/class/GuildChannel) doesn't include thread channels.

Done.

File: packages/discord.js/src/managers/GuildChannelManager.js [hunk 1]
```diff
@@ -317,7 +317,7 @@ class GuildChannelManager extends CachedManager {
    * Obtains one or more guild channels from Discord, or the channel cache if they're already available.
    * @param {Snowflake} [id] The channel's id
    * @param {BaseFetchOptions} [options] Additional options for this fetch
-   * @returns {Promise<?GuildChannel|Collection<Snowflake, GuildChannel>>}
+   * @returns {Promise<?(GuildChannel|ThreadChannel)|Collection<Snowflake, GuildChannel>>}
```
Comment (almeidx) [inline packages/discord.js/src/managers/GuildChannelManager.js:320]:
```suggestion
   * @returns {Promise<?(GuildChannel|ThreadChannel|Collection<Snowflake, GuildChannel>)>}
```

**Review by almeidx — CHANGES_REQUESTED** [review]

File: packages/discord.js/src/managers/GuildChannelManager.js [hunk 2]
```diff
@@ -317,7 +317,7 @@ class GuildChannelManager extends CachedManager {
    * Obtains one or more guild channels from Discord, or the channel cache if they're already available.
    * @param {Snowflake} [id] The channel's id
    * @param {BaseFetchOptions} [options] Additional options for this fetch
-   * @returns {Promise<?GuildChannel|Collection<Snowflake, GuildChannel>>}
+   * @returns {Promise<?(GuildChannel|ThreadChannel|Collection<Snowflake, GuildChannel>)>}
```
Comment (Jiralite) [inline packages/discord.js/src/managers/GuildChannelManager.js:320]:
```suggestion
   * @returns {Promise<?GuildChannel|ThreadChannel|Collection<Snowflake, ?GuildChannel>>}
```

File: packages/discord.js/typings/index.d.ts [hunk 3]
```diff
@@ -3376,7 +3376,7 @@ export class GuildChannelManager extends CachedManager<Snowflake, GuildBasedChan
   public create(options: GuildChannelCreateOptions): Promise<TextChannel>;
   public createWebhook(options: WebhookCreateOptions): Promise<Webhook>;
   public edit(channel: GuildChannelResolvable, data: GuildChannelEditOptions): Promise<GuildChannel>;
-  public fetch(id: Snowflake, options?: BaseFetchOptions): Promise<NonThreadGuildBasedChannel | null>;
+  public fetch(id: Snowflake, options?: BaseFetchOptions): Promise<GuildBasedChannel | null>;
   public fetch(id?: undefined, options?: BaseFetchOptions): Promise<Collection<Snowflake, NonThreadGuildBasedChannel>>;
```
Comment (Jiralite) [inline packages/discord.js/typings/index.d.ts:3380]:
```suggestion
  public fetch(id?: undefined, options?: BaseFetchOptions): Promise<Collection<Snowflake, NonThreadGuildBasedChannel | null>>;
```

**Review by Jiralite — COMMENTED** [review]
Fetching multiple channels actually calls `ChannelManager#_add()` so it appears even channels in the collection may be `null`.

**Review by kyranet — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**